library/java/src/io/envoyproxy/envoymobile/engine/JvmBridgeUtility.java

```diff
@@ -0,0 +1,75 @@
+package io.envoyproxy.envoymobile.engine;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.envoyproxy.envoymobile.engine.types.EnvoyHTTPCallbacks;
+
+/**
+ * Class to assist with passing types from native code over the JNI. Currently supports
+ * HTTP headers.
+ */
+class JvmBridgeUtility {
```

Comment:
Could we add a couple unit tests for the behaviors expected for this class?

Reply:
Yes, though if you're okay with it, I'd prefer to do it in a separate PR to keep things rolling forward on main.